src/App.jsx

```diff
@@ -1,7 +1,21 @@
 import { useState } from "react";
+import { BrowserRouter, Routes, Route } from "react-router-dom";
+import Homepage from "./pages/Homepage";
+import ArticlesByTopic from "./pages/ArticlesByTopic";
+import SingleArticle from "./pages/SingleArticle";
+import AllArticles from "./pages/AllArticles";
 
 function App() {
-  return <></>;
+  return (
+    <BrowserRouter>
+      <Routes>
+        <Route path="/" element={<Homepage />} />
+        <Route path="/allArticles" element={<AllArticles />} />
+        <Route path="/article/:topic" element={<ArticlesByTopic />} />
```

Comment:
Following a ticket system - you normally just add the code for the ticket you're completing. If it doesnt ask for other pages, you don't normally add them